pym/bob/generators/QtCreatorGenerator.py

```diff
@@ -141,6 +141,14 @@ def generateQtProject(package, destination, updateOnly, projectName, includeDirs
     if not os.path.exists(destination):
         os.makedirs(destination)
 
+    # setup list of exclude packages
+    excludePackages = []
+    for e in excludes:
+        exp = re.compile(e)
+        for name,path in OrderedDict(sorted(dirs, key=lambda t: t[1])).items():
+            if exp.match(name):
+                excludePackages.append(name)
+
```

Comment:
I would move this to `getCheckoutDirs`. You can simply skip a excluded directory and not add it to the dict.